dlrs/main/classes/RollupSummariesSelector.cls

```diff
@@ -363,37 +426,164 @@ public class RollupSummariesSelector {
-              LookupRollupSummary2__mdt.RelationshipField__c,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .toSOQL()
-        )
-      );
+      if (this.forEdit) {
+        return RollupSummary.toList(
+          this.getLookupRollupSummaries(
+            newQueryFactory()
+              .setCondition('DeveloperName in :tmpVar1 and Active__c = true')
+              .addOrdering(
+                LookupRollupSummary2__mdt.ParentObject__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .addOrdering(
+                LookupRollupSummary2__mdt.RelationshipField__c,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>{ uniqueNames }
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = new List<RollupSummary>();
+        List<SObject> records = new List<SObject>();
+        for (String uniqueName : uniqueNames) {
+          SObject record = getLookupRollupSummaryByDeveloperNameOrId(
+            uniqueName
+          );
+          if (record != null && (Boolean) record.get('Active__c')) {
+            records.add(record);
+          }
+        }
+
+        return sortSummaries(
+          RollupSummary.toList(records),
+          'ParentObject__c',
+          'RelationshipField__c'
+        );
+      }
     }
 
     /**
      * Select all
      **/
     public List<RollupSummary> selectAll() {
-      return RollupSummary.toList(
-        Database.query(
-          newQueryFactory()
-            .addOrdering(
-              LookupRollupSummary2__mdt.Label,
-              fflib_QueryFactory.SortOrder.ASCENDING
-            )
-            .toSOQL()
-        )
-      );
+      if (this.forEdit) {
+        return RollupSummary.toList(
+          this.getLookupRollupSummaries(
+            newQueryFactory()
+              .addOrdering(
+                LookupRollupSummary2__mdt.Label,
+                fflib_QueryFactory.SortOrder.ASCENDING
+              )
+              .toSOQL(),
+            new List<Object>()
+          )
+        );
+      } else {
+        List<RollupSummary> summaries = RollupSummary.toList(
+          this.getLookupRollupSummaries()
+        );
+
+        return sortSummaries(summaries, 'Label', null);
+      }
+    }
+
+    protected virtual List<SObject> getLookupRollupSummaries(
+      String queryString,
+      List<Object> tmpVars
+    ) {
+      Object tmpVar1 = tmpVars.size() > 0 ? tmpVars[0] : null;
+      Object tmpVar2 = tmpVars.size() > 1 ? tmpVars[1] : null;
+
+      return Database.query(queryString);
+    }
+
+    protected virtual List<SObject> getLookupRollupSummaries() {
+      return LookupRollupSummary2__mdt.getAll().values();
+    }
+
+    protected virtual SObject getLookupRollupSummaryByDeveloperNameOrId(
+      String developerNameOrId
+    ) {
+      return LookupRollupSummary2__mdt.getInstance(developerNameOrId);
```

Comment:
@afawcett We talked about how verifying that this method returns null and does not throw an exception. I just verified in my dev org that it does indeed return null and does not throw an exception.

Nothing to do here - just an FYI since we talked about this one.